keel-web/src/test/kotlin/com/netflix/spinnaker/keel/rest/ApplicationControllerTests.kt

```diff
@@ -127,14 +166,11 @@ internal class ApplicationControllerTests : JUnit5Minutests {
 
     context("application with delivery config exists") {
       before {
-        deliveryConfigRepository.store(deliveryConfig)
-        environments.values.forEach { env ->
-          env.resources.forEach { res ->
-            resourceRepository.store(res)
-            resourceRepository.appendHistory(ResourceCreated(res))
-          }
+        combinedRepository.upsertDeliveryConfig(deliveryConfig)
+        // these events are required because Resource.toResourceSummary() relies on events to determine resource status
```

Comment:
Is it possible to just mock the call to status so you can return whatever you want w/o having to persist the events? if not, that's fine.

Reply:
Good catch. Yep, totally! Will change.

Reply:
Hmm, actually, the way the tests are currently written, we depend on a "real" `ResourceRepository` (in-memory, but not a mock) that's wired into `ApplicationService` to return the resource summaries, so I'll have to leave it like this.